From a8b8566c4f71e894879e33996e5a29f64c6c44fd Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Mon, 3 Mar 2014 00:00:46 +0100
Subject: [PATCH] Extend write-built-using to also generate the
 ${extra:Built-Using} substvar.

This makes it possible to track packages mentioned in the EXTRA_PACKAGES
variable (currently set to "bf-utf-source syslinux"). Thanks to Ansgar
Burchardt for the reports (Closes: #696418, #700026).

Note: Unknown packages are ignored, so architecture-specific packages
shouldn't be a problem.
---
 build/util/write-built-using | 7 +++++++
 debian/changelog             | 4 ++++
 debian/control               | 2 +-
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/build/util/write-built-using b/build/util/write-built-using
index 97beb13..c9c9c73 100755
--- a/build/util/write-built-using
+++ b/build/util/write-built-using
@@ -6,3 +6,10 @@ cat build/tmp/*/built-using.txt \
     | tr "\n" "," \
     | sed 's/,/, /g;s/, $//g' >> debian/debian-installer.substvars
 echo >> debian/debian-installer.substvars
+
+EXTRA_PACKAGES="bf-utf-source syslinux not-existing"
+echo -n "extra:Built-Using=" >> debian/debian-installer.substvars
+for package in $EXTRA_PACKAGES; do
+  dpkg-query -f '${source:Package} (= ${source:Version}), ' -W $package 2>/dev/null
+done >> debian/debian-installer.substvars
+echo >> debian/debian-installer.substvars
diff --git a/debian/changelog b/debian/changelog
index 83c3382..1a0e3b7 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -20,6 +20,10 @@ debian-installer (20140210) UNRELEASED; urgency=medium
   * build/Makefile: Move the drop_lang definition to the top of the file to
     deal with incompatible changes in make 3.82 leading to the following
     error: “recipe commences before first target” (Closes: #720719).
+  * Extend write-built-using to also generate the ${extra:Built-Using}
+    substvar, making it possible to track packages mentioned in the
+    EXTRA_PACKAGES variable (currently set to "bf-utf-source syslinux").
+    Thanks to Ansgar Burchardt for the reports (Closes: #696418, #700026).
 
   [ Robert Millan ]
   * Enable kfreebsd-10 builds.
diff --git a/debian/control b/debian/control
index 61f8311..3d21dee 100644
--- a/debian/control
+++ b/debian/control
@@ -164,7 +164,7 @@ Build-Depends:
 
 Package: debian-installer
 Architecture: any
-Built-Using: ${initrd:Built-Using}
+Built-Using: ${initrd:Built-Using}, ${extra:Built-Using}
 Depends: ${misc:Depends}
 Description: Debian installer
  This package currently only contains some documentation for the Debian
-- 
1.9.0

